File: Qsn_6/shadow3D.cpp
int shadow3D(vector<vector<int>>& grid) {
        int N = grid.size();
        int ans = 0;

        for (int i = 0; i < N;  ++i) {
            int row = 0;
            int col = 0; 
            for (int j = 0; j < N; ++j) {
                if (grid[i][j] > 0) ans++;
                row = max(row, grid[i][j]);
                col = max(col, grid[j][i]);
            }
            ans += row + col;
        }

        return ans;
    }

File: Qsn_6/shadow3D.py
def shadow3D(x):
    N = len(x)
    ans = 0

    for i in range(N):
        row = 0
        col = 0
        for j in range(N):
            if x[i][j]: ans += 1  # top shadow
            row = max(row, x[i][j])
            col = max(col, x[j][i])

        ans += row + col

    return ans
print(shadow3D([[1,2],[3,4]]))
